lib/common/strings.c

```diff
@@ -1169,6 +1173,20 @@ pcmk__strcmp(const char *s1, const char *s2, uint32_t flags)
         }
     }
 
+    /* If this flag is set, return 0 if either (or both) of the input strings
+     * are "*".  If neither one is, we need to continue and compare them
+     * normally.
+     */
+    if (pcmk_is_set(flags, pcmk__str_star_matches)) {
+        if (s1 == NULL || s2 == NULL) {
+            return 0;
+        }
```

Comment:
Why do we want to match NULL?

Reply:
If one of the strings is NULL, but pcmk__str_null_matches was not given, we'll get a segfault when the strcmp is called.

Reply:
The way it's currently implemented, sure :) The question is why implement it that way -- having star_matches also match null seems counter-intuitive

Reply:
Fixed.